src/data/polynomial/algebra_map.lean

```diff
@@ -293,7 +293,13 @@ variables [algebra S R] [algebra S A'] [algebra S B']
 /-- Version of `aeval` for defining algebra homs out of `polynomial R` over a smaller base ring
   than `R`. -/
 def aeval_tower (f : R →ₐ[S] A') (x : A') : R[X] →ₐ[S] A' :=
-{ commutes' := λ r, by simp [algebra_map_apply],
```

Comment:
The reason `eval₂_C` doesn't apply here is because the expression looks like `↑(eval₂ f x)`, where the coercion goes from `polynomial R → S` to itself, arising from `algebra.id`. That is definitely not what we want to happen, let alone implicitly, so now I know that my `has_coe_t` suggestion was wrong.

Reply:
OK so let's stick with `has_lift_t` for now.